src/Core/Extensible.php

```diff
@@ -18,7 +17,18 @@
  */
 trait Extensible
 {
-    use CustomMethods;
+    use CustomMethods {
```

Comment:
The problem is that the code would error before it got to this point.

This line is called to get $obj and would fail.

```php
$obj = $config['index'] !== null ?
                    $this->{$config['property']}[$config['index']] :
                    $this->{$config['property']};
```

$config['property'] would be `extension_instances` so we need to force that property to be populated by that point.